math/pythagoras.cpp

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright 2020 @author beqakd
+ * @file
+ *
+ * A basic implementation of 3d distance between two point.
+ */
+
+#include <cassert>   // For testing(assert)
+#include <cmath>     // For std::pow(), std::sqrt() and so on
+#include <iostream>  // For io operations
+
+/**
+ * The distance between two points in a three dimensional - 3D - coordinate
+ * system. For this we have struct point where we save coordinates of our points
+ * and later on calculate the distance as mentioned.
+ */
+class Distance {
+ public:
+    /**
+    * Point struct to save 3D coordinates.
+    */
+    struct Point {
+        int x;
+        int y;
+        int z;
+    };
+
+    // Our two points to calculate distance.
+    Point a;
+    Point b;
+
+    /**
+    * Constructor of this class.
+    * @param point_fir 3D coordinates of first point.
+    * @param point_sec 3d coordinates of second point.
+    */
+    Distance(const int point_fir[], const int point_sec[]) {
```

Comment:
why making things too complex when we can achive same functionality with even same speed and accuracy with simple things.

Reply:
Cool. Got your point.  You want to have variable initialized in test method i mean point a and point b right?

Reply:
For this i still need to have Point struct